src/components/Explorer/departures.ts

```diff
@@ -33,15 +33,33 @@ const getBestCandidateDepartureTime = (
 ) => {
     let bestCandidate: null | number = null;
     let bestDifference = Infinity;
-    candidates.forEach((candidate) => {
-        const nextEnhancedArrival = enhancedArrivals.find((t) => t >= candidate)!;
-        const difference = nextEnhancedArrival - candidate;
-        if (difference < bestDifference) {
-            bestDifference = difference;
-            bestCandidate = candidate;
+    if (candidates.length > 0) {
+        candidates.forEach((candidate) => {
+            const nextEnhancedArrival = enhancedArrivals.find((t) => t >= candidate)!;
+            const difference = nextEnhancedArrival - candidate;
+            if (difference < bestDifference) {
+                bestDifference = difference;
+                bestCandidate = candidate;
+            }
+        });
+        return bestCandidate;
+    }
+    // if we don't have candidates from the baseline, we need to pick a time based on enhancedArrivals
+    let firstArrival: null | number = null;
+    let secondArrival: null | number = null;
+    enhancedArrivals.forEach((arrival) => {
+        if (arrival > 32400 && !firstArrival) {
+            firstArrival = arrival;
+        } else if (arrival > 32400 && firstArrival && !secondArrival) {
+            secondArrival = arrival;
         }
     });
-    return bestCandidate;
+    if (firstArrival && secondArrival) {
+        // return midpoint between first 2 arrivals after 9am
+        return (firstArrival + secondArrival) / 2;
+    }
+    // if we still haven't found anything, default to 9am
+    return 32400;
```

Comment:
We use this several times throughout the function, could we define it somewhere near the top of this block of code and reuse the constant?

```ts
const targetArrival = 9 * 60 * 60; // 9 AM
```